app/assets/javascripts/components/CartedProducts.jsx

```diff
@@ -0,0 +1,65 @@
+var CartedProducts = React.createClass({
+  getInitialState: function(){
+     return {
+       initialItems: this.props.carted_products,
+       carted_products: []
+     }
+  },
+  componentWillMount: function(){
+    this.setState({carted_products: this.state.initialItems})
+  },
+  updateQuantity: function(val, id) {
+    console.log(val, "from parent && id: ", id);
+    this.props.carted_products.map((carted_product) =>
+      (carted_product.id == id) ? (
+        carted_product.quantity = val
+      ) : (
+        carted_product
+      )
+    );
+    $.ajax({
+      type: "PATCH",
+      url: "/api/carted_products/" + id + "/" + val,
+      contentType: "application/json; charset=utf-8",
+      dataType: "json",
+      success: function(result){
+        console.log(result);
+      }
+    });
+    this.setState({carted_products: this.state.carted_products});
```

Comment:
In a future pr, this should probably be moved inside the `success` callback, right?